controllers/application_controller.go

```diff
@@ -101,21 +110,34 @@ func (r *ApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			}
 		}
 		// Create or Update resources (configmap and HR) if not already installed
-		if err = r.CreateOrUpdateResources(application, ctx); err != nil {
+		if err = r.CreateOrUpdateResources(ctx, application); err != nil {
 			v1.AppErrorStatus(application, err.Error())
-			if err1 := r.patchStatus(ctx, application); err1 != nil {
-				return ctrl.Result{RequeueAfter: r.RequeueInterval}, err1
+			if patchErr := r.patchStatus(ctx, application); patchErr != nil {
+				return ctrl.Result{RequeueAfter: r.RequeueInterval}, patchErr
 			}
 			return ctrl.Result{}, err
 		}
 
 		// At this point the Helm Release can be reconciled
-		err = r.reconcileHRStatus(application, ctx)
-		if err1 := r.patchStatus(ctx, application); err1 != nil {
-			log.Error(err1, "Error updating application status")
-			return ctrl.Result{}, err1
+		err := r.reconcileHelmReleaseStatus(ctx, application)
+		if patchErr := r.patchStatus(ctx, application); patchErr != nil {
+			log.Error(patchErr, "Error updating application status")
+			return ctrl.Result{}, patchErr
+		}
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		// Retrieve the pod status
+		err = r.reconcilePodStatus(ctx, application)
```

Comment:
may not reconcile pod status when helm release is still getting the helm chart ready for instance and still did not update the helm resources. The reason being that you may provide older deployment status here if there is an older one.

Reply:
This isn't really a problem, because the HelmRelease has control of the `Ready` condition which dictates the true "health" of the application. In other words, as long as the HelmRelease is in a non-ready state, the Application will not be either.

Reply:
so, the pod status is reconciled only if the helm release is in ready status? Sorry its hard to understand from reading github code

Reply:
So, wouldnt you be providing an incorrect analysis condition if you are referring to an older deployment?

Reply:
You're right. 

Resolved by ca5f866